modules/logging/filters.go

```diff
@@ -185,15 +187,117 @@ func (m IPMaskFilter) Filter(in zapcore.Field) zapcore.Field {
 	return in
 }
 
+type ActionType string
+
+const (
+	// Replace value(s) of query parameter(s).
+	ReplaceType ActionType = "replace"
+	// Delete query parameter(s).
+	DeleteType ActionType = "delete"
+)
+
+type queryFilterAction struct {
+	// `replace` to replace the value(s) associated with the parameter(s) or `delete` to remove them entirely.
+	Type ActionType `json:"type"`
+	// The name of the query parameter.
+	Parameter string `json:"parameter"`
+	// The value to use as replacement if the action is `replace`.
+	Value string `json:"value,omitempty"`
```

Comment:
Style thing, we generally use a newline between fields when there's comments, to space it out a bit

```suggestion
	// `replace` to replace the value(s) associated with the parameter(s) or `delete` to remove them entirely.
	Type ActionType `json:"type"`

	// The name of the query parameter.
	Parameter string `json:"parameter"`

	// The value to use as replacement if the action is `replace`.
	Value string `json:"value,omitempty"`
```